crates/subspace-networking/src/node.rs

```diff
@@ -602,4 +602,9 @@ impl Node {
     pub fn on_peer_info(&self, callback: HandlerFn<NewPeerInfo>) -> HandlerId {
         self.shared.handlers.new_peer_info.add(callback)
     }
+
+    /// Callback is called when a peer is disconnected.
+    pub fn on_disconnected_peer(&self, callback: HandlerFn<PeerId>) -> HandlerId {
+        self.shared.handlers.disconnected_peer.add(callback)
+    }
```

Comment:
We don't need the `peer connected` event for this task because receiving `peer_info` implies that a peer was connected. We can add a new event later when we need it.

Reply:
I know we don't, but we likely will for some reason. It is inconsistent to introduce API that allows you to know that someone disconnected without knowing that someone connected in the first place. API looks ugly/incomplete because of this. Even if we don't need it today, we need to think about the future a bit and how API looks as well.